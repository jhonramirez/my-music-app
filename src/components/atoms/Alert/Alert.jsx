import React from "react";
import "./Alert.scss";

export const Alert = ({ message }) => {
  return <div id="snackbar">{message}</div>;
};
