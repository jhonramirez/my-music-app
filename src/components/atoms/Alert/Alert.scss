#snackbar {
  visibility: hidden;
  min-width: 250px; 
  margin-left: -125px; 
  color: rgb(255, 255, 255);
  background-color: rgb(151, 144, 138);
  text-align: center; 
  border-radius: 30px; 
  padding: 10px; 
  position: fixed; 
  z-index: 1; 
  left: 50%; 
  bottom: 30px; 

}

.show {
  visibility: visible !important; 
  -webkit-animation: fadein 0.5s, fadeout 0.5s 2.5s;
  animation: fadein 0.5s, fadeout 0.5s 2.5s;
}
