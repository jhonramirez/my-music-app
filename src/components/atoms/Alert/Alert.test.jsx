import { render, screen } from "@testing-library/react";
import { Alert } from "./Alert";
import "@testing-library/jest-dom";

describe("Pruebas en <Alert/>", () => {
  const { container } = render(<Alert message={"proof"} id={"snackbar"} />);
  const AlertElement = screen.getByText("proof");

  test("Render Alert y match con el snapshot", () => {
    expect(AlertElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("El id del Alert debe ser snackbar", () => {
    expect(AlertElement).toHaveAttribute("id", "snackbar");
  });
});
