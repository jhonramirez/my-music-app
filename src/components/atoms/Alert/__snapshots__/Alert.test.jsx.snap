// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Alert/> Render Alert y match con el snapshot 1`] = `
<div>
  <div
    id="snackbar"
  >
    proof
  </div>
</div>
`;
